packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -96,6 +96,9 @@ export const SelectPopover = styled( Ariakit.SelectPopover )`
 	border-radius: 2px;
 	background: ${ COLORS.theme.background };
 	border: 1px solid ${ COLORS.theme.foreground };
+	z-index: 9999; // Ensure the popover is on top
+	position: absolute !important;
+	max-height: 400px;
```

Comment:
Setting the max-height to `400px` is in line with what the V1 does, but we should do it in way suggested by Ariakit, using the `--popover-available-height` CSS variable so that the popover will be even shorter if there isn't available screen estate:

```CSS
max-height: min(var(--popover-available-height, 400px), 400px);
overflow: auto;
overscroll-behavior: contain;
```

Reply:
Opened https://github.com/WordPress/gutenberg/pull/62821 to work on the select popover fixes separately